### PR TITLE
Joe Jiang

### PR DESCRIPTION
**Pull request by ninesky00**
Was stuck on figuring out an error on buildings.renters. 
Only completed up to iteration 3.

### CODE REVIEW

File: lib/apartment.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Apartment
+    attr_reader :number, :monthly_rent, :bathrooms, :bedrooms, :renter
```
Comment (mikedao) [inline lib/apartment.rb:2]: Watch your indentation, should be two spaces

File: lib/building.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        @units << unit
+    end
+
+    def average_rent
+        total_rent = @units.inject(0) {|sum, unit| sum += unit.monthly_rent}
+        (total_rent.to_f / 2).round(1)
+    end
+
+    def renters
+        @renters = @units.map {|unit| unit.renter.name}
```
Comment (mikedao) [inline lib/building.rb:20]: This method should just return the collection, not store it in an instance variable. This method does not handle the case when there are units with no renters. Use the multi line enumerable syntax.

File: lib/building.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    end
+
+    def rented_units
+       @rented_units = @units.select {|unit| unit.renter != nil} 
+    end
+
+    def renter_with_highest_rent
+        rented_units.max_by {|unit| unit.monthly_rent}
```
Comment (mikedao) [inline lib/building.rb:28]: This returns the unit, not the renter.

**Review by mikedao — COMMENTED** [review]
Score: 2.0

Make sure you are reading the specification carefully, and you are returning what the specification is asking for.

File: lib/building.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    end
+
+    def units_by_number_of_bedrooms
+        @units.group_by {|unit| unit.bedrooms}
```
Comment (mikedao) [inline lib/building.rb:32]: This returns apartment objects, which we do not want

File: lib/apartment.rb (hunk 1 above)
Comment (ninesky00) [reply lib/apartment.rb:2]: I think I left my tab spaces default as 4 from vs code, I should change it to 2?

File: lib/building.rb (hunk 2 above)
Comment (ninesky00) [reply lib/building.rb:20]: completely overlooked this, I was able to fix it with multi line enumerable?(does below count?) 
 def renters
        renters = []
        @units.each do |unit|
            if unit.renter != nil    
                renters << unit.renter.name
            end
        end
        renters
    end

File: lib/building.rb (hunk 3 above)
Comment (ninesky00) [reply lib/building.rb:28]: changed to   
def renter_with_highest_rent
        highest_unit = rented_units.max_by {|unit| unit.monthly_rent}
        highest_unit.renter
  end
I believe this returns the desired output

File: lib/building.rb (hunk 4 above)
Comment (ninesky00) [reply lib/building.rb:32]: I knew I made a mistake, didn't have time to go over it. 
I believe the below suffices the requirement?
def units_by_number_of_bedrooms
        hash = Hash.new
        @units.each do |unit|
            hash[unit.bedrooms] = []
        end
        @units.each do |unit|
            if hash[unit.bedrooms] 
                hash[unit.bedrooms] << unit.number
            end
        end
        hash
    end

**Comment by ninesky00** [comment]
Thank you for the corrections Mike!

**Pull request closed without merging**